enterprise/internal/completions/streaming/limiter.go

```diff
@@ -130,19 +147,34 @@ func (r *rateLimiter) TryAcquire(ctx context.Context) (err error) {
 	return nil
 }
 
-func userKey(userID int32) string {
+func userKey(userID int32, scope RateLimitScope) string {
+	if scope == RateLimitScopeCodeCompletion {
+		return fmt.Sprintf("user:%d:code_completion_requests", userID)
+	}
```

Comment:
if we made RateLimitScope string and not int, we wouldn't need so many switches in this code, and could do something like

```suggestion
func userKey(userID int32, scope RateLimitScope) string {
		return fmt.Sprintf("user:%d:%s_requests", userID, scope)
```

Reply:
done!